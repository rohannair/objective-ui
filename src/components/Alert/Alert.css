@import '../../styles/mixins';
@define-extend alert {
  align-items: center;
  border: 1px solid;
  border-radius: 2px;
  display: flex;
  flex-direction: row;
  font-size: 0.85rem;
  height: 3rem;
  line-height: 3rem;
  margin-bottom: 20px;
  padding: 0 15px;
}

.body {
  margin-left: 10px;
}

.close {
  margin-left: auto;
  cursor: pointer;
}

.success {
  @extend alert;
  @mixin success;
}

.info {
  @extend alert;
  @mixin info;
}

.warn {
  @extend alert;
  @mixin warning;
}

.danger {
  @extend alert;
  @mixin danger;
}
