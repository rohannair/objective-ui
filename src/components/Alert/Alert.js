import React, { PropTypes } from 'react';
import styles from './Alert.css';
import classNames from 'classnames/bind';

let cx = classNames.bind(styles);

const Alert = p => {
  const classname = cx({
    [styles[p.type]]: true
  });


  return (
    <div className={classname}>
      { getIcon(p.type) }
      <div className={styles.body}>
        { p.children }
      </div>
      <div className={styles.close}>
        <i className={'zmdi zmdi-close'} onClick={p.close}/>
      </div>
    </div>
  );
};

Alert.proptypes = {
  type: PropTypes.oneOf(['info', 'danger', 'warn', 'success']),
  children: PropTypes.string,
  close: PropTypes.func
};

const iconMap = {
  info: 'info-outline',
  danger: 'alert-polygon',
  warn: 'alert-triangle',
  success: 'check-circle'
};
const getIcon = (type) => <i className={`zmdi zmdi-${iconMap[type]}`} />;

export default Alert;
