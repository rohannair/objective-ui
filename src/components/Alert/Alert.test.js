import React from 'react'
import Alert from './index.js'

describe('Test suite for Alert component', () => {
  it('Alert should exist', () => {
    let wrapper = shallow(<Alert/>)
    expect(wrapper).toMatchSnapshot()
  })
})

