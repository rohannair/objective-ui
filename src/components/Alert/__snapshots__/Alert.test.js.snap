exports[`Test suite for Alert component Alert should exist 1`] = `
<div
  className="undefined">
  <i
    className="zmdi zmdi-undefined" />
  <div />
</div>
`;
